jax/_src/linear_util.py

```diff
@@ -248,11 +253,30 @@ def fun_name(f):
   except:
     return str(f)
 
-def wrap_init(f, params=None) -> WrappedFun:
+class TracingDebugInfo(NamedTuple):
+  # Packages up trace/staging-time debug info about a func and its parameters,
+  # formed just before staging to a jaxpr and read in trace-time error messages.
+  traced_for: str             # e.g. 'jit', 'scan', etc
+  func_src_info: str | None   # e.g. f'{fun.__name__} at {filename}:{lineno}'
+  arg_names: tuple[str, ...]  # e.g. ('args[0]', ... )
+  # e.g. ('[0]', '[1]', ...)
+  result_paths_thunk: Callable[[], tuple[str, ...]] | None
+
+  @classmethod
+  def from_jaxpr(cls, jaxpr: core.ClosedJaxpr) -> TracingDebugInfo | None:
```

Comment:
It might be worth calling this `from_closed_jaxpr` just to be clear?

Reply:
At some point JAX started to have both `Jaxpr` and `ClosedJaxpr`. Now the majority of the usage is `ClosedJaxpr`, yet we still refer to that as `jaxpr` in function and variable names.  I feel that we should unify these two, at the very least to reduce confusion, and then it makes sense to standardize on `ClosedJaxpr`.